packages/key4hep-external-stack/package.py

```diff
@@ -0,0 +1,68 @@
+#!/usr/bin/env python3
+
+from datetime import datetime
+
+from spack.pkg.k4.key4hep_stack import Key4hepPackage, install_setup_script
+
+from common import *
+
+
+class Key4hepExternalStack(BundlePackage, Key4hepPackage):
+    """Bundle package that contains a basic external software stack upon which
+    Key4hep can be built
+
+    The packages in this base stack are mainly
+    """
+
+    homepage = "https://cern.ch/key4hep"
+
+    version(datetime.today().strftime("%Y-%m-%d"))
+
+    # this bundle package installs a custom setup script, so
+    # need to add the install phase (which normally does not
+    # exist for a bundle package)
+    phases = ["install"]
+
+    # Some generally useful development tools
+    depends_on("cmake")
+    depends_on("ninja")
+    depends_on("python")
+    depends_on("gdb")
+    depends_on("catch2@3:")
+    depends_on("boost")
+    depends_on("py-pytest")
+
+    # general hep packages
+    depends_on("root")
```

Comment:
`+root7` is not a default one and is probably needed in some places, others may also be needed to prevent a dependency from needing another ROOT.

Reply:
But this should be covered by including the common `packages.yaml`, I think.

Reply:
I wonder if changing the requirements in the common `packages.yaml` and affecting the external stack will be a problem or not :thinking:

Reply:
Good question. I think it shouldn't since the ones specified in the external environment take precedence. However, if the requirements in the common packages are underspecified, then building a `key4hep-stack` on top of the external environment might lead to some packages being rebuilt for the latter because not all variants are present.

This might be something that could be checked in CI (maybe in a follow up PR): concretize this external environment and either the release or nightly environment and see if the packages that are common to both get concretized the same.